spring-kafka-docs/src/main/antora/modules/ROOT/pages/testing.adoc

```diff
@@ -674,3 +674,22 @@ The transactional id is provided in case you wish to use a different `MockProduc
 If you are using producers in a multi-threaded environment, the `BiFunction` should return multiple producers (perhaps thread-bound using a `ThreadLocal`).
 
 IMPORTANT: Transactional `MockProducer`+++s+++ must be initialized for transactions by calling `initTransaction()`.
+
+When using the `MockProducer`, if you do not want to close the producer after each send, then you can provide a custom `MockProducer` implementation that overrides the `close` method that does not call the `close` method from the super class.
+This is convenient for testing, when verifying multiple publishing on the same producer without closing it.
+
+Here is an example:
+
+[source, java]
+----
+@Bean
+ProducerFactory<String, String> mockProducerFactory() {
+    return new MockProducerFactory<>(() -> new MockProducer<>(false, new StringSerializer(), new StringSerializer()) {
```

Comment:
This sample is a bit misleading.
When we really use a `MockProducer`, we perform some verification afterwards.
Therefore an instance for this `MockProducer` must be shared and accessible from the unit tests.
I probably won't mind to have it as a bean in the test configuration and then just refer to it from this supplier for `MockProducerFactory`.
Otherwise a configuration like this would always lead to a new `MockProducer` instance every time we call `KafkaTemplate.send()`.
Just because the `createProducer()` always call this `Supplier.get()`.